libr/core/cmd_debug.c

```diff
@@ -1571,6 +1572,34 @@ static int r_debug_heap(RCore *core, const char *input) {
 	return true;
 }
 
+static bool get_bin_info(RCore *core, const char *file, ut64 baseaddr, int mode, bool symbols_only, const char *symname) {
+	int fd;
+	if ((fd = r_io_fd_open (core->io, file, R_PERM_R, 0)) != -1) {
+		RBinOptions opt = { 0 };
+		opt.fd = fd;
+		opt.sz = r_io_fd_size (core->io, fd);
+		opt.baseaddr = baseaddr;
+		RBinFile *obf = r_bin_cur (core->bin);
+		if (r_bin_open_io (core->bin, &opt)) {
+			int action = R_CORE_BIN_ACC_ALL & ~R_CORE_BIN_ACC_INFO;
+			if (symbols_only || symname) {
+				action = R_CORE_BIN_ACC_SYMBOLS;
+			}
+			if (symname) {
+				r_cons_grep (symname);
```

Comment:
is there any test for this? In other code I'm using `r_cons_grep_process`, though I'm not sure how they are different. Also, shall you use `r_cons_push` + `r_cons_pop`?